### DISCUSSION TITLE
Early scope definition stage at a meal delivery startup.

### DISCUSSION THREAD

**Discussion by ghost** [saleor/saleor#14101] [General] [answered]
I am defining the requirements for a mobile first, hyperlocal meal delivery startup. It's not a marketplace.

I'm a business person even before this startup, on the product management side.

So we are looking into using a headless CMS and Saleor looks great! However we have the need for some features which I believe don't natively ship with Saleor and I wanted to get an idea if they can *without much pain* be integrated by our own devs.

1. OTP authentication- I've looked around and it seems like we can implement this.

2. Delivery slot selection- The user picks the time of delivery, we are hyperlocal so no shipping, just last mile delivery.

3. A calendar based meal planning system- The user can pre-select meal delivery on a calendar.(No. 5 has some relevance here also)

4. A knowledge base- with media that unlocks for the items they purchase. This seems doable to me.

5. Edits to the order flow- when the user payment is confirmed by the gateway, a job order will be sent out to a third party manufacturing app for the kitchen to pick, prepare, pack and label the items which will then go to -> 'ready for delivery state' -> 'out for delivery' -> 'fulfilled'. I need to record this on Saleor too.

What I want to know is-

1. If these integrations are possible.

2. How difficult would some of these features be to implement. I'm working with experienced, thoughtful devs who are new to Saleor but work with python, node.js, typescript, ...

3. What kind of migration challenges(to a newer version) I could face down the line.

I'm not asking how to build these features, just how it would be to integrate them with the Saleor CMS without breaking anything.

Edit: Cleaned up a bit.

**Comment by timuric** [comment, marked as answer]
Hey @rlstyne Saleor is designed to be composable with other systems, the type of integrations you are describing are very common among majority of projects we have seen. You should be able to achieve all of that using our graphql api (we strongly discourage modifying saleor core). If you have use cases that you were unable to solve please open a new discussion with enough business context for the community to be able to advise.

**Reply by ghost** [reply]
Hey @timuric, our developers are testing some of the use cases right now. I'll open a thread if we hit any roadblocks, thanks!

**Comment by peteror** [comment]
HI - I'm working on something very similar. Most of these are not part of Saleor, but you can easily integrate with things your team builds. 
We ended up building a separate component for this in Python (we have delivery slot selection, OTP authentication, an external delivery system - with similar statuses to yours - and cash on delivery for payment.) using the Saleor GraphQL API. We also have a mobile app our customers use as a front-end.

**Reply by ghost** [reply]
Hi @peteror, thanks for replying. We have a fully working functional prototype for a single store made with no code tools and firebase(auth, CMS). We used this to test the market.
However, our business plan involves a multi-store model. We want to build the MVP with multi store support servicing different sets of postal codes with each store having some unique products and inventory levels. We are confident Saleor can accommodate most if not all our use cases.
